packages/alfa-css/src/value/transform/scale.ts

```diff
@@ -4,46 +4,89 @@ import { Parser } from "@siteimprove/alfa-parser";
 import { Function as CSSFunction } from "../../syntax/index.js";
 
 import { List } from "../collection/index.js";
-import { Number } from "../numeric/index.js";
+import { Number, Percentage } from "../numeric/index.js";
 import type { Resolvable } from "../resolvable.js";
 
 import { Function } from "./function.js";
 
 const { map, either } = Parser;
 
+// We cannot easily use Resolvable.Resolved because Percentage may resolve to
+// anything depending on the base, here we want to keep them as percentages.
+type ToCanonical<T extends Number | Percentage<"percentage">> = T extends Number
+  ? Number.Canonical
+  : T extends Percentage
+    ? Percentage.Canonical
+    : Number.Canonical | Percentage.Canonical;
+
 /**
+ * Class representing a transformation that resizes an element in either 2D or 3D space.
+ * Is parsed from either a scale() or scale3d() transform function or a value of the scale property.
+ *
  * @public
  */
-export class Scale
+export class Scale<
+    X extends Number.Canonical | Percentage.Canonical =
+      | Number.Canonical
+      | Percentage.Fixed<"percentage">,
+    Y extends Number.Canonical | Percentage.Canonical =
+      | Number.Canonical
+      | Percentage.Fixed<"percentage">,
+    Z extends Number.Canonical | Percentage.Canonical =
+      | Number.Canonical
+      | Percentage.Fixed<"percentage">,
+  >
   extends Function<"scale", false>
   implements Resolvable<Scale.Canonical, never>
 {
-  public static of(x: Number, y: Number): Scale {
-    return new Scale(x.resolve(), y.resolve());
+  public static of<
+    X extends Number.Canonical | Percentage.Canonical,
+    Y extends Number.Canonical | Percentage.Canonical,
+    Z extends Number.Canonical | Percentage.Canonical,
+  >(x: X, y: Y, z?: Z): Scale<X, Y, Z>;
+
+  public static of<
+    X extends Number | Percentage<"percentage">,
+    Y extends Number | Percentage<"percentage">,
+    Z extends Number | Percentage<"percentage">,
+  >(x: X, y: Y, z?: Z): Scale<ToCanonical<X>, ToCanonical<Y>, ToCanonical<Z>>;
```

Comment:
```suggestion
  public static of<
    X extends Number | Percentage<"percentage">,
    Y extends Number | Percentage<"percentage">,
  >(x: X, y: Y): Scale<ToCanonical<X>, ToCanonical<Y>, never>;
```
Seems like that overload should only have 2 parameters, since we already have an overload for 3D scale 🤔 
(i.e., with an optional `z`, this already covers the 3-values overload which is not needed)